### PR TITLE
[DPE-4875] Apply sysctl requirements if possible

### PR DESCRIPTION
**Pull request by Mehdi-Bendriss**
## Issue
This PR implements [DPE-4875](https://warthogs.atlassian.net/browse/DPE-4875), namely:
- check sysctl requirements 
- apply them if needed and possible
- check them again for verification

[DPE-4875]: https://warthogs.atlassian.net/browse/DPE-4875?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by reneradoi — APPROVED** [review]
nit: iterate over the tuples of prop, val and the operator (>, <) to get rid of the multiple if's. But I'll leave it up to you if you want that or not.

**Comment by Mehdi-Bendriss** [comment]
Thanks @reneradoi ! I thought about it a bit after reading your comment, and given that the missing requirements message varies (depending on the comparison operator) it will end up with as much lines as the current one with slightly less readability

File: lib/charms/opensearch/v0/opensearch_relation_provider.py [hunk 1]
```diff
@@ -59,7 +59,7 @@
 
 # Increment this PATCH version before using `charmcraft publish-lib` or reset
 # to 0 if you are raising the major API version
-LIBPATCH = 3
```
Comment (phvalguima) [inline lib/charms/opensearch/v0/opensearch_relation_provider.py:62]: Is this correct?

File: lib/charms/opensearch/v0/opensearch_distro.py [hunk 2]
```diff
@@ -452,24 +452,30 @@ def normalize_allocation_exclusions(exclusions: Union[List[str], Set[str], str])
...
+            missing_requirements.append(f"{prop} should be {val}")
 
-        tcp_retries = int(
-            subprocess.getoutput("sysctl net.ipv4.tcp_retries2").split("=")[-1].strip()
-        )
-        if tcp_retries > 5:
-            missing_requirements.append("net.ipv4.tcp_retries2 should be 5")
+        prop, val = "net.ipv4.tcp_retries2", "5"
+        if self._run_cmd(f"sysctl -n {prop}") > val and not apply(prop, val):
+            missing_requirements.append(f"{prop} should be at most {val}")
```
Comment (phvalguima) [inline lib/charms/opensearch/v0/opensearch_distro.py:478]: What will we do about `fs.file-max`?

**Review by phvalguima — COMMENTED** [review]
We should also clean up our docs from cloud-init. We will not need it if we will try setting sysctl via charm. Main reason:
1. If we are running as VMs, then we will be able to set all sysctl parameters
2. If we are running within LXC, then we cannot set them either way: via charm or via cloud-init; we need an external intervention to set it on the host itself

So, these docs are now redundant.

File: lib/charms/opensearch/v0/opensearch_relation_provider.py (hunk 1 above)
Comment (Mehdi-Bendriss) [reply lib/charms/opensearch/v0/opensearch_relation_provider.py:62]: Yes - the Libpatch on charmhub is 1, i'm a bit unsure why we had 3 set in the first place

File: lib/charms/opensearch/v0/opensearch_distro.py (hunk 2 above)
Comment (Mehdi-Bendriss) [reply lib/charms/opensearch/v0/opensearch_distro.py:478]: This is already set in the snap through `ulimits` see [here](https://github.com/canonical/opensearch-snap/blob/main/scripts/wrappers/start.sh#L8) and [here](https://github.com/canonical/opensearch-snap/blob/main/scripts/wrappers/sys/set-sys-config.sh#L29-L32)

**Comment by Mehdi-Bendriss** [comment]
@phvalguima The reason why I chose to keep as is the docs is to because the charm only sets "the minimal" best values - if the user deployed using cloud-init better values our charm logic should/will not change those. 
So I prefer to allow the user some flexibility to apply better values than we do ourselves.

**Comment by phvalguima** [comment]
> @phvalguima The reason why I chose to keep as is the docs is to because the charm only sets "the minimal" best values - if the user deployed using cloud-init better values our charm logic should/will not change those.
So I prefer to allow the user some flexibility to apply better values than we do ourselves.

@Mehdi-Bendriss then in this case we should move the session down in our README, as the cloud init is not mandatory anymore. Maybe adding a `Customizing` session, but def. cloud-init is not part of `Usage` anymore.

**Review by phvalguima — APPROVED** [review]
Approving as we will review the sysctl docs and testing on a next PR

**Pull request merged**